call-home.sh

```diff
@@ -0,0 +1,286 @@
+    json_message_map["product_family"]=$PERCONA_PRODUCT_FAMILY
+
+    metrics_map["pillar_version"]=$PERCONA_PRODUCT_VERSION
+    # TODO:
+    # On my Ubuntu, the value of /etc/issue is "Ubuntu 22.04.3 LTS \n \l"
+    # Escape characters cause problems in Json, so remove them.
+    # But maybe the better idea is to use values as provided and then
+    # sanitize the whole json_message string after create_json_message()
+    # by escaping all escape characters?
+    # (see the comment at the end of collect_data_for_report())
+    metrics_map["OS"]=${PERCONA_OPERATING_SYSTEM//\\/}
+    metrics_map["deployment"]=$PERCONA_DEPLOYMENT_METHOD
+}
+
+
+# {
+#    "reports":[
+#       {
+#          "id":"13f5fc62-35b4-4716-b3e6-96c761fc204d",
+#          "createTime":"2023-09-15T13:36:53+03:00",
+#          "instanceId":"6e5ff5d4-5617-11ee-8c99-0242ac120002",
+#          "product_family":"PERCONA_PRODUCT_FAMILY_MYSQL",
+#          "metrics":[
+#             {
+#                "key":"pillar_version",
+#                "value":"8.0.30"
+#             },
+#             {
+#                "key":"OS",
+#                "value":"Ubuntu 22.04"
+#             },
+#             {
+#                "key":"deployment",
+#                "value":"PACKAGE"
+#             }
+#          ]
+#       }
+#    ]
+# }
+
+# Let's create a single JSON message here.
+create_json_message()
+{  
+    json_message="{"
+      json_message+="\"reports\":["
+        json_message+="{"
+    
+          # report header
+          for key in "${!json_message_map[@]}"
+          do
+              m="$(printf "\"%s\" : \"%s\"" "$key" "${json_message_map[$key]}")"
+              json_message+="${m},"
+          done
+      
+          # metrics
+          local first_metric=1
+          json_message+="\"metrics\":["
+      
+            for key in "${!metrics_map[@]}"
+            do
+                if [[ $first_metric -ne 1 ]];
+                then
+                  json_message+=","
+                fi
+
+                json_message+="{"
+                m1="$(printf "\"key\" : \"%s\"" "$key")"
+                m2="$(printf "\"value\" : \"%s\"" "${metrics_map[$key]}")"
+                json_message+="${m1},"
+                json_message+="${m2}"
+                json_message+="}"
+                first_metric=0
+            done
+      
+          json_message+="]"   # metrics
+        json_message+="}"   # reports
+      json_message+="]"   # reports
+
+    json_message+="}"
+
+    # TODO:
+    # sanitize the message (in case of any escape characters, escape them)
+    # json_message=${json_message//\\/\\\\}
+}
+
+send_json_message()
+{
+    (curl -X POST --connect-timeout $PERCONA_SEND_TIMEOUT --header 'Content-Type: application/json' --location $PERCONA_TELEMETRY_URL --data "$json_message") &>/dev/null
+}
+
+# Check options passed in.
+while getopts "h f:v:s:d:i:j:u:t:" OPTION
+do
+    case $OPTION in
+        h)
+            usage
+            exit_script 1
```

Comment:
in case the script content will be used as is in package post-install - do not forget to change all exit codes to 0

Reply:
Let's wait for Evgeniy's opinion